### PR TITLE
Re-enable GL testing in CI

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Connections**

This is a follow up to #2504, currently in draft state to get CI to run. Will rebase once it is merged in.

**Description**
GL testing back on CI!

**Testing**
It is.

### CODE REVIEW

**Review by kvark — APPROVED** [review]

**Comment by cwfitzgerald** [comment]
Re-requesting review on this, I've rebased against merged master. Same deal commits are free-standing.

Most should be pretty uncontroversial, but I added a downlevel flag, and added a validation canary to wgpu-hal, which you should take a look at.

**Review by kvark — APPROVED** [review]

File: wgpu/tests/clear_texture.rs [hunk 1]
```diff
@@ -239,20 +239,28 @@ fn single_texture_clear_test(
...
 fn clear_texture_tests(
     ctx: &TestingContext,
     formats: &[wgpu::TextureFormat],
     supports_1d: bool,
     supports_3d: bool,
 ) {
     for &format in formats {
+        let desc = format.describe();
+        let rounded_width = round_up(64, desc.block_dimensions.0 as u32);
+        let rounded_height = round_up(64, desc.block_dimensions.1 as u32);
```
Comment (kvark) [inline wgpu/tests/clear_texture.rs:255]: why are we rounding?
WebGPU allows creating any texture sizes regardless of the block size

**Review by kvark — CHANGES_REQUESTED** [review]

File: wgpu/tests/clear_texture.rs (hunk 1 above)
Comment (cwfitzgerald) [reply wgpu/tests/clear_texture.rs:255]: What how does that work with vk requires textures a multiple if the block size?

Comment (cwfitzgerald) [reply wgpu/tests/clear_texture.rs:255]: Per https://gpuweb.github.io/gpuweb/#dom-gpudevice-createtexture at least currently, images need to be the block size.

> descriptor.[size](https://gpuweb.github.io/gpuweb/#dom-gputexturedescriptor-size).[width](https://gpuweb.github.io/gpuweb/#extent3d-width) must be multiple of [texel block width](https://gpuweb.github.io/gpuweb/#texel-block-width).

> descriptor.[size](https://gpuweb.github.io/gpuweb/#dom-gputexturedescriptor-size).[height](https://gpuweb.github.io/gpuweb/#extent3d-height) must be multiple of [texel block height](https://gpuweb.github.io/gpuweb/#texel-block-height).

Comment (kvark) [reply wgpu/tests/clear_texture.rs:255]: oh right, hmm

**Review by kvark — APPROVED** [review]

**Pull request merged**